orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/domain/ApplyResponse.java

```diff
@@ -5,23 +5,30 @@
 
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.JsonNode;
 
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class ApplyResponse {
 
     @JsonProperty("return")
-    private List<Map<String, Object>> result;
+    private List<Map<String, JsonNode>> result;
 
-    public Iterable<Map<String, Object>> getResult() {
+    public Iterable<Map<String, JsonNode>> getResult() {
         return result;
     }
 
-    public void setResult(List<Map<String, Object>> result) {
+    public void setResult(List<Map<String, JsonNode>> result) {
         this.result = result;
     }
 
     public String getJid() {
-        return result.get(0).get("jid").toString();
+        if (result != null && result.size() > 0) {
+            Map<String, JsonNode> resultMap = result.get(0);
```

Comment:
This result.get(0) looks suspicious. Are we sure we always want the first results?

Reply:
Unfortunately salt api always returns with result list. The reason is it is possible to send more than one command per request, but we always send one command per request from cloudbreak. Anyway I just put some asserts around this get and I didn't modified this logic.